rasterio/io.py

```diff
@@ -105,7 +105,7 @@ class WindowMethodsMixin(object):
     properties: `transform`, `height` and `width`
     """
 
-    def window(self, left, bottom, right, top, boundless=False, precision=6):
+    def window(self, left, bottom, right, top, boundless=True, precision=6):
```

Comment:
@perrygeo here's an API change. In the context of all the other work I did, a boundless Window makes much more sense.